src/CoreLib/DSetCSharpLinq.fs

```diff
@@ -0,0 +1,104 @@
+(*---------------------------------------------------------------------------
+    Copyright 2013 Microsoft
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.                                                      
+
+    File: 
+        DSetCSharpLinq.fs
+  
+    Description: 
+        CSharp Linq APIs for DSet<'U>
+
+    Author:																	
+        Jin Li, Principal Researcher
+        Microsoft Research, One Microsoft Way
+        Email: jinl at microsoft dot com
+    Date:
+        Sept. 2015
+    
+ ---------------------------------------------------------------------------*)
+namespace Prajna.Api.CSharp.Linq
```

Comment:
Do we need "CSharp" in the namespace? This makes it seem that CSharp is non-native for Prajna. And "Api" is kind of implied. This will be a core, popular API, so it would be nice to have a short name. Why not "Prajna.Linq"?